tests/v2/validation/provisioning/k3s/custom_cluster_test.go

```diff
@@ -76,24 +77,39 @@ func (c *CustomClusterProvisioningTestSuite) SetupSuite() {
 }
 
 func (c *CustomClusterProvisioningTestSuite) TestProvisioningK3SCustomCluster() {
-	nodeRoles0 := []string{
-		"--etcd --controlplane --worker",
-	}
+	var etcdNode machinepools.NodeRoles
+	etcdNode.Etcd = true
 
-	nodeRoles1 := []string{
-		"--etcd --controlplane",
-		"--worker",
-	}
+	var controlPlaneNode machinepools.NodeRoles
+	controlPlaneNode.ControlPlane = true
 
-	nodeRoles2 := []string{
-		"--etcd",
-		"--controlplane",
-		"--worker",
-	}
+	var workerNode machinepools.NodeRoles
+	workerNode.Worker = true
+
+	var allRolesNode machinepools.NodeRoles
+	allRolesNode.Etcd = true
+	allRolesNode.ControlPlane = true
+	allRolesNode.Worker = true
+
+	var etcdControlPlaneNode machinepools.NodeRoles
+	etcdControlPlaneNode.Etcd = true
+	etcdControlPlaneNode.ControlPlane = true
```

Comment:
Maybe we can move these to a file called `provisioning/provisioning.go` or use the existing `provisioning/config.go`, instantiate as structs there, and import them to the tests, would it be more readable? Seems like we do the same things for RKE1 custom tests

Reply:
yeah kind of how we do with machine pools.

Reply:
okay, I've added this to the latest push; please resolve if the solution looks good.